models/Document/Editable/Date.php

```diff
@@ -94,7 +92,6 @@ public function frontend()
      */
     public function getDataForResource()
     {
-        $this->checkValidity();
```

Comment:
I don't think this makes sense here because this class doesn't implement the `checkValidity` method, which means the one from `parent` is called which only returns `true`.

Reply:
yes, good catch!